header-rewrite-filter/header_processor.cc

```diff
@@ -53,6 +53,46 @@ namespace HeaderRewriteFilter {
         return absl::OkStatus();
     }
 
+    absl::Status AppendHeaderProcessor::parseOperation(std::vector<absl::string_view>& operation_expression, std::vector<absl::string_view>::iterator start) {
+        if (operation_expression.size() < Utility::APPEND_HEADER_MIN_NUM_ARGUMENTS) {
+            return absl::InvalidArgumentError("not enough arguments for set-header");
+        }
+
+        // TODO: remove
+        start++;
+
+        // parse key and call setKey
+        try {
+            const absl::string_view key = operation_expression.at(2); // TODO: use start
+            setKey(key);
+        } catch (const std::exception& e) {
+            // should never happen, range is checked above
+            return absl::InvalidArgumentError("error parsing header key");
```

Comment:
Nit: would be good to print the message of the exception, to make debugging easier. E.g.
```
return absl::InvalidArgumentError(std::string("error parsing header key: ") + e.what());
```